lib/fastlane/plugin/firebase_app_distribution/client/firebase_app_distribution_api_client.rb

```diff
@@ -0,0 +1,148 @@
+      end
+
+      def upload_binary(app_id, binary_path)
+        connection.post(binary_upload_url(app_id), File.open(binary_path).read) do |request|
+          request.headers["Authorization"] = "Bearer " + auth_token
+        end
+      rescue Faraday::ResourceNotFound
+        UI.crash!("#{ErrorMessage::INVALID_APP_ID}: #{app_id}")
+      rescue Errno::ENOENT
+        UI.crash!("#{ErrorMessage::APK_NOT_FOUND}: #{binary_path}")
+      end
+
+      # Uploads the binary
+      #
+      # Returns the release_id on a successful release.
+      # Returns nil if unable to upload.
+      def upload(app_id, binary_path)
+        upload_token = get_upload_token(app_id, binary_path)
+        upload_status_response = get_upload_status(app_id, upload_token)
+        if upload_status_response.success?
+          UI.success("This APK/IPA has been uploaded before. Skipping upload step.")
+        else
+          UI.message("This APK has not been uploaded before.")
+          MAX_POLLING_RETRIES.times do
+            if upload_status_response.success?
+              UI.success("Uploaded APK/IPA Successfully!")
+              break
+            elsif upload_status_response.in_progress?
+              sleep(POLLING_INTERVAL_SECONDS)
+            else
+              UI.message("Uploading the APK/IPA.")
+              upload_binary(app_id, binary_path)
+            end
+            upload_status_response = get_upload_status(app_id, upload_token)
+          end
+          unless upload_status_response.success?
+            UI.error("It took longer than expected to process your APK/IPA, please try again.")
+            return nil
+          end
+        end
+        upload_status_response.release_id
+      end
+
+      # Gets the upload status for the app release.
+      def get_upload_status(app_id, app_token)
+        begin
+          response = connection.get(upload_status_url(app_id, app_token)) do |request|
+            request.headers["Authorization"] = "Bearer " + auth_token
+          end
+        rescue Faraday::ResourceNotFound
+          UI.crash!("#{ErrorMessage::INVALID_APP_ID}: #{app_id}")
+        end
+        return UploadStatusResponse.new(response.body)
+      end
+
+      private
+
+      def v1_apps_url(app_id)
+        "/v1alpha/apps/#{app_id}"
+      end
+
+      def release_notes_create_url(app_id, release_id)
+        "#{v1_apps_url(app_id)}/releases/#{release_id}/notes"
+      end
+
+      def binary_upload_url(app_id)
+        "/app-binary-uploads?app_id=#{app_id}"
+      end
+
+      def upload_status_url(app_id, app_token)
+        "#{v1_apps_url(app_id)}/upload_status/#{app_token}"
+      end
+
+      def upload_token_format(app_id, project_number, binary_hash)
+        CGI.escape("projects/#{project_number}/apps/#{app_id}/releases/-/binaries/#{binary_hash}")
+      end
+
+      def connection
+        @connection ||= Faraday.new(url: Helper::FirebaseAppDistributionHelper::BASE_URL) do |conn|
```

Comment:
BASE_URL has moved right? I'm guessing this would break if you tried to run it.

Reply:
nice catch